### ISSUE TITLE
JavaCGOpal CLI generates non-unique IDs for callables in resolved call graph.

### ISSUE THREAD

**Issue by cg122** [fasten-project/fasten#159]
## Describe the bug
JavaCGOpal generates non-unique ID for callables in resolved call graph.

## To Reproduce
1. Using example in **(https://github.com/fasten-project/fasten-maven-plugin/tree/master/src/test/resources-its/eu/fasten/maven/CheckMojoIT/test1)**
2. Compile the code and generate JAR file (test1-test.jar)
3. Use JavaCGOpal CLI to generate merged graph with dependency JAR file (commons-text-1.9.jar)
```
java -jar javacg-opal-0.1.2-SNAPSHOT-with-dependencies.jar -a test1-test.jar -d commons-text-1.9.jar -s -ma CHA -o result.json
```
In the generated result.json_merged, two different callables have the same ID "0".

```json
{
    "product": "test1-test",
    "nodes": 0,
    "forge": "",
    "generator": "",
    "version": "",
    "cha": {
        "externalTypes": {
            "/org.apache.commons.text/WordUtils": {
                "access": "",
                "methods": {"3": {
                    "metadata": {},
                    "uri": "/org.apache.commons.text/WordUtils.capitalize(%2Fjava.lang%2FString)%2Fjava.lang%2FString"
                }},
                "final": false,
                "superInterfaces": [],
                "sourceFile": "",
                "superClasses": []
            },
            "/java.lang/Object": {
                "access": "",
                "methods": {"2": {
                    "metadata": {},
                    "uri": "/java.lang/Object.%3Cinit%3E()VoidType"
                }},
                "final": false,
                "superInterfaces": [],
                "sourceFile": "",
                "superClasses": []
            }
        },
        "internalTypes": {"/null/Test1": {
            "access": "public",
            "methods": {
                "0": {
                    "metadata": {
                        "access": "public",
                        "last": 3,
                        "defined": true,
                        "first": 3
                    },
                    "uri": "/null/Test1.%3Cinit%3E()%2Fjava.lang%2FVoidType"
                },
                "1": {
                    "metadata": {
                        "access": "public",
                        "last": 7,
                        "defined": true,
                        "first": 7
                    },
                    "uri": "/null/Test1.foo()%2Fjava.lang%2FString"
                }
            },
            "final": false,
            "superInterfaces": [],
            "sourceFile": "/Test1.java",
            "superClasses": ["/java.lang/Object"]
        }},
        "resolvedTypes": {"//commons-text-1.9$/org.apache.commons.text/WordUtils": {
            "access": "public",
            "methods": {"0": {
                "metadata": {
                    "access": "public",
                    "last": 410,
                    "defined": true,
                    "first": 410
                },
                "uri": "/org.apache.commons.text/WordUtils.capitalize(%2Fjava.lang%2FString)%2Fjava.lang%2FString"
            }},
            "final": false,
            "superInterfaces": [],
            "sourceFile": "org/apache/commons/text/WordUtils.java",
            "superClasses": ["/java.lang/Object"]
        }}
    },
    "graph": {
        "internalCalls": [],
        "externalCalls": [
            [
                "0",
                "2",
                {"1": {
                    "receiver": "/java.lang/Object",
                    "type": "invokespecial",
                    "line": 3
                }}
            ],
            [
                "1",
                "3",
                {"2": {
                    "receiver": "/org.apache.commons.text/WordUtils",
                    "type": "invokestatic",
                    "line": 7
                }}
            ]
        ],
        "resolvedCalls": [[
            "1",
            "0",
            {"2": {
                "receiver": "/org.apache.commons.text/WordUtils",
                "type": "invokestatic",
                "line": 7
            }}
        ]]
    },
    "timestamp": 0
}
```

## Expected behavior
Global ID generation scheme for both artifact and all provided dependencies in local merge.

## Additional context
JavaCGOpa CLI used isl compiled from 'develop'  branch.

**Comment by amottier** [comment]
In my not help at all but it might worth noticing that you are not using the latest version available in the `develop` branch.

In fact the current version is using `-m` as replacement for `-s` (see https://github.com/fasten-project/fasten/tree/develop/analyzer/javacg-opal#arguments and in source code https://github.com/fasten-project/fasten/blob/develop/analyzer/javacg-opal/src/main/java/eu/fasten/analyzer/javacgopal/Main.java#L122).

Maybe your local `develop` branch is behind `origin/develop` and you might want to try to update it and do again your test (again I have no clue whether or not updating will include modification that can address your issue).

**Comment by cg122** [comment]
Thanks for pointing this out. I will try the latest version and see how it works.

**Comment by ashkboos** [comment]
It should be fixed by bbecb85
